hack/Jenkins/W2L/setup.sh

```diff
@@ -0,0 +1,241 @@
+
+
+# Verify we can get the remote node to respond to _ping
+if [ $ec -eq 0 ]; then
+	reply=`curl -s http://$ip:2357/_ping`
+	if [ "$reply" != "OK" ]; then
+		ec=1
+		echo "ERROR: Failed to get OK response from Linux node at $ip:2357. It may be down."
+		echo "       Try re-running this CI job, or ask on #docker-dev or #docker-maintainers"
+		echo "       to see if the node is up and running."
+		echo
+		echo "Regular ping output for remote host below. It should reply. If not, it needs restarting."
+		ping $ip
+	else
+		echo "INFO: The Linux nodes outer daemon replied to a ping. Good!"
+	fi 
+fi
+
+# Get the version from the remote node. Note this may fail if jq is not installed.
+# That's probably worth checking to make sure, just in case.
+if [ $ec -eq 0 ]; then
+	remoteVersion=`curl -s http://$ip:2357/version | jq -c '.Version'`
+	echo "INFO: Remote daemon is running docker version $remoteVersion"
+fi
+
+# Compare versions. We should really fail if result is no 1. Output at end of script.
+if [ $ec -eq 0 ]; then
+	uniques=`docker version | grep Version | /usr/bin/sort -u | wc -l`
+fi
+
+# Make sure we are in repo
+if [ $ec -eq 0 ]; then
+	if [ ! -d hack ]; then
+		echo "ERROR: Are you sure this is being launched from a the root of docker repository?"
+		echo "       If this is a Windows CI machine, it should be c:\jenkins\gopath\src\github.com\docker\docker."
+                echo "       Current directory is `pwd`"
+		ec=1
+	fi
+fi
+
+# Get the commit has and verify we have something
+if [ $ec -eq 0 ]; then
+	export COMMITHASH=$(git rev-parse --short HEAD)
+	echo INFO: Commmit hash is $COMMITHASH
+	if [ -z $COMMITHASH ]; then
+		echo "ERROR: Failed to get commit hash. Are you sure this is a docker repository?"
+		ec=1
+	fi
+fi
+
+# Redirect to a temporary location. Check is here for local runs from Jenkins machines just in case not
+# in the right directory where the repo is cloned. We also redirect TEMP to not use the environment
+# TEMP as when running as a standard user (not local system), it otherwise exposes a bug in posix tar which
+# will cause CI to fail from Windows to Linux. Obviously it's not best practice to ever run as local system...
+if [ $ec -eq 0 ]; then
+	export TEMP=/c/CI/CI-$COMMITHASH
+	export TMP=$TMP
+	/usr/bin/mkdir -p $TEMP  # Make sure Linux mkdir for -p
+fi
+
+# Tidy up time
+if [ $ec -eq 0 ]; then
+	echo INFO: Deleting pre-existing containers and images...
+	# Force remove all containers based on a previously built image with this commit
+	! docker rm -f $(docker ps -aq --filter "ancestor=docker:$COMMITHASH") &>/dev/null
+
+	# Force remove any container with this commithash as a name
+	! docker rm -f $(docker ps -aq --filter "name=docker-$COMMITHASH") &>/dev/null
+
+	# Force remove the image if it exists
+	! docker rmi -f "docker-$COMMITHASH" &>/dev/null
+	
+	# This SHOULD never happen, but just in case, also blow away any containers
+	# that might be around. 
+	! if [ ! `docker ps -aq | wc -l` -eq 0 ]; then
```

Comment:
thanks!

On Mon, Jan 4, 2016 at 3:58 PM, John Howard (Microsoft) <
notifications@github.com> wrote:

> In hack/Jenkins/W2L/setup.sh
> <https://github.com/docker/docker/pull/18461#discussion_r48798328>:
>
> > +
> > +# Tidy up time
> > +if [ $ec -eq 0 ]; then
> > +	echo INFO: Deleting pre-existing containers and images...
> > +	# Force remove all containers based on a previously built image with this commit
> > +	! docker rm -f $(docker ps -aq --filter "ancestor=docker:$COMMITHASH") &>/dev/null
> > +
> > +	# Force remove any container with this commithash as a name
> > +	! docker rm -f $(docker ps -aq --filter "name=docker-$COMMITHASH") &>/dev/null
> > +
> > +	# Force remove the image if it exists
> > +	! docker rmi -f "docker-$COMMITHASH" &>/dev/null
> > +	
> > +	# This SHOULD never happen, but just in case, also blow away any containers
> > +	# that might be around.
> > +	! if [ ! `docker ps -aq | wc -l` -eq 0 ]; then
>
> @jfrazelle <https://github.com/jfrazelle> Just in case :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/18461/files#r48798328>.
>